# Mapping short and long reads onto a reference genome
Once you set your environment you can execute now the rest of the pipeline

## Beginning of the pipeline:
This section will deal with mapping short and long reads (srRNAseq and lrRNAseq). This module can be used to map reads as a first step and/or re-run using available annotation both from annotations already available or produced along the rest of the pipeline (*e.g.:* using the prediction or the `BRAKER` step).

Create a folder where to run all the analyses:
```bash
mkdir -p $DIR && cd $DIR
```

1. Execute `compleasm`, an implementation of BUSCO genome that employs `miniprot`. This will serve you to understand the amount of BUSCO genes, "at best", that can be found on your assembly. You want the final annotation to approach these statistics as much as possible.

```bash
compleasm.py run -t16 -l $BUSCODB -a $DATADIR/$SPECIES.$CHRNAME.fasta -o $SPECIES.$CHRNAME.Busco.$BUSCODB
```

2. Now use a mapper that can splice your short reads. In this workshop, I used `STAR` but any other srRNAseq would work.
I like STAR because it returns a set of statistics to evaluate how the mapping was performed. I also like the `two-pass` strategy, where `STAR` uses the first round of mapping to discover splice sites and uses them to map reads a second time.
But first, you have to create an `index`. Look at `STAR --help` to check available options.

 ```bash
STAR --runMode genomeGenerate --genomeDir $SPECIES.$CHRNAME.STAR.Index --runThreadN $THREADS \
	--genomeFastaFiles $DATADIR/$SPECIES.$CHRNAME.fasta --genomeChrBinNbits 8 --genomeSAindexNbases 11
```
3. Now it's the time to map your reads. You can use a `for` loop to map all the SRA srRNAseq libraries

```bash
for SRA in $srSRAS; do
	STAR --genomeDir $SPECIES.$CHRNAME.STAR.Index --readFilesIn $DATADIR/${SRA}_1.extracted.fastq.gz $DATADIR/${SRA}_2.extracted.fastq.gz --outFilterType BySJout \
		--outSAMattributes All --outSAMtype BAM Unsorted --readFilesCommand zcat \
		--runThreadN 16 --alignEndsType Local --outStd Log --limitBAMsortRAM 7900000000 \
		--alignSJoverhangMin 10 --outFileNamePrefix $SPECIES.RNA.$CHRNAME.$SRA.
done
```
4. You can now merge or sort the alignments

```bash
ls -1 $SPECIES.RNA.$CHRNAME.*.Aligned.out.bam > BAM.list
cat BAM.list

samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.RNA.$CHRNAME.Aligned.merged.bam
samtools sort -@ 20 $SPECIES.RNA.$CHRNAME.Aligned.merged.bam > $SPECIES.RNA.$CHRNAME.Aligned.sortedByCoord.out.bam
```

5. The same for the long-reads (lrRNAseq)
   
```bash
for SRA in $longREADS; do
	winnowmap -ax splice $CHR $DATADIR/$SRA.extracted.fastq.gz | samtools view -@ 30 -O BAM > $SPECIES.$SRA.lonrRNA.bam
	samtools view -@ 30 -O BAM $SPECIES.$SRA.lonrRNA.bam | samtools sort -@ 30 -n > $SPECIES.$SRA.lonrRNA.$CHRNAME.bam
done
```

5. Merge and sort

```bash
ls -1 $SPECIES.*.lonrRNA.$CHRNAME.bam > BAM.list
cat BAM.list

samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam
samtools sort -@ 20 $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam > $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam
samtools index -@ 20 $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam
```
