# Set your variable
```
THREADS=
DATADIR=
BUSCODIR=
PFAM=
SWISSPROTDB=
TRINITY_HOME=


#SPECIES=Atal
#SPECIES=Epro
SPECIES=Sgre
```

## This are the variable needed to annotate *Arabidopsis taliana*
```
SPECIES=Atal
DIR=Tutorial.Arabidopsis
srSRAS='SRR5956435 SRR5956436'
longREADS='SRR23291376 SRR23291378'
LONGTYPE=pacbio
ASSEMBLY=$DATADIR/GCF_000001735.4_TAIR10.1_genomic.fna
TARGETCHR='NC_003070.9'
CHRNAME=chr1
CHR=$DATADIR/$SPECIES.$CHRNAME.fasta
ANNREF=$DATADIR/$SPECIES.$CHRNAME.gene.ncbi.gff
BRAKERGTF=$DATADIR/$SPECIES.braker_utr.$CHRNAME.gff3
STRAND='--rf'
LIBTYPE='first'
BUSCODB=embryophyta_odb10
```

#--------------------------------------------------
#|Results from dataset embryophyta_odb10: proteome|
#--------------------------------------------------
#|C:27.1%[S:14.4%,D:12.7%],F:0.3%,M:72.6%,n:1614  |
#|437   Complete BUSCOs (C)                       |
#|232   Complete and single-copy BUSCOs (S)       |
#|205   Complete and duplicated BUSCOs (D)        |
#|5     Fragmented BUSCOs (F)                     |
#|1172  Missing BUSCOs (M)                        |
#|1614  Total BUSCO groups searched               |
#--------------------------------------------------
#
#--------------------------------------------------
#|Results from dataset embryophyta_odb10: Chr1    |
#--------------------------------------------------
#|S:27.01%,D:0.19%,F:0.93%,I:0.00%,M:71.87%,n:1614|
#|436   Complete and single-copy BUSCOs (S)       |
#|3     Complete and duplicated BUSCOs (D)        |
#|0     Incomplete (I)                            |
#|15    Fragmented BUSCOs (F)                     |
#|1160   Missing BUSCOs (M)                       |
#|1614  Total BUSCO groups searched               |
#--------------------------------------------------

#################################################################################
elif [[ $SPECIES == 'Epro' ]]; then
#################################################################################
# Entomobrya proxima
SPECIES=Epro
DIR=Tutorial.Springtail
srSRAS='SRR15910090'
longREADS='SRR15910089'
LONGTYPE=pacbio
ASSEMBLY=$DATADIR/GCA_029691765.1_ASM2969176v1_genomic.fna
TARGETCHR='CM056168.1'
CHRNAME=chr1
CHR=$DATADIR/$SPECIES.$CHRNAME.fasta
ANNREF=$DATADIR/$SPECIES.$CHRNAME.gene.maker.gff
BRAKERGTF=$DATADIR/$SPECIES.braker_utr.$CHRNAME.gff3
STRAND='--rf'
LIBTYPE='first'
BUSCODB=arthropoda_odb10

#--------------------------------------------------
#|Results from dataset arthropoda_odb10: proteome |
#--------------------------------------------------
#|C:32.1%[S:28.9%,D:3.2%],F:1.9%,M:66.0%,n:1013   |
#|325   Complete BUSCOs (C)                       |
#|293   Complete and single-copy BUSCOs (S)       |
#|32    Complete and duplicated BUSCOs (D)        |
#|19    Fragmented BUSCOs (F)                     |
#|669   Missing BUSCOs (M)                        |
#|1013  Total BUSCO groups searched               |
#--------------------------------------------------

#--------------------------------------------------
#|Results from dataset arthropoda_odb10: Chr1     |
#--------------------------------------------------
#|S:30.40%,D:0.10%,F:1.09%,I:0.00%,M:68.41%,n:1013|
#|308   Complete and single-copy BUSCOs (S)       |
#|1     Complete and duplicated BUSCOs (D)        |
#|0     Incomplete (I)                            |
#|11    Fragmented BUSCOs (F)                     |
#|693   Missing BUSCOs (M)                        |
#|1013  Total BUSCO groups searched               |
#--------------------------------------------------

#################################################################################
elif [[ $SPECIES == 'Sgre' ]]; then
#################################################################################
# Schistocerca gregaria
SPECIES=Sgre
DIR=Tutorial.DesertLocust
srSRAS='SRR15423966 SRR15423964'
longREADS='SRR17978912'
LONGTYPE=pacbio
ASSEMBLY=$DATADIR/GCF_023897955.1_iqSchGreg1.2_genomic.fna
TARGETCHR='NC_064930.1'
CHRNAME=chr11
CHR=$DATADIR/$SPECIES.$CHRNAME.fasta
ANNREF=$DATADIR/$SPECIES.$CHRNAME.gene.ncbi.gff
BRAKERGTF=$DATADIR/$SPECIES.braker_utr.$CHRNAME.gff3
STRAND='--rf'
LIBTYPE='first'
BUSCODB=insecta_odb10

#--------------------------------------------------
#|Results from dataset insecta_odb10              |
#--------------------------------------------------
#|C:1.7%[S:1.0%,D:0.7%],F:0.2%,M:98.1%,n:1367     |
#|22	Complete BUSCOs (C)                       |
#|13	Complete and single-copy BUSCOs (S)       |
#|9	    Complete and duplicated BUSCOs (D)        |
#|3	    Fragmented BUSCOs (F)                     |
#|1342	Missing BUSCOs (M)                        |
#|1367	Total BUSCO groups searched               |
#--------------------------------------------------

#--------------------------------------------------
#|Results from dataset insecta_odb10: Chr11       |
#--------------------------------------------------
#|S:1.10%,D:0.00%,F:0.29%,I:0.00%,M:98.61%,n:1367 |
#|15    Complete and single-copy BUSCOs (S)       |
#|0     Complete and duplicated BUSCOs (D)        |
#|0     Incomplete (I)                            |
#|4     Fragmented BUSCOs (F)                     |
#|1348  Missing BUSCOs (M)                        |
#|1367  Total BUSCO groups searched               |
#--------------------------------------------------

#################################################################################
fi

mkdir -p $DIR && cd $DIR

#. $APPS2/compleasm/venv/bin/activate
#$APPS2/compleasm/compleasm.py run -t16 -l $BUSCODB -a $DATADIR/$SPECIES.$CHRNAME.fasta -o $SPECIES.$CHRNAME.Busco.$BUSCODB
#deactivate

if [[ ! -s $SPECIES.$CHRNAME.STAR.Index ]]; then
	echo -e "###### `date`: Index for $SPECIES assembly."
	echo 'STAR --runMode genomeGenerate --genomeDir $SPECIES.$CHRNAME.STAR.Index --runThreadN $THREADS --genomeFastaFiles $DATADIR/$SPECIES.$CHRNAME.fasta --genomeChrBinNbits 8 --genomeSAindexNbases 11'
	time STAR --runMode genomeGenerate --genomeDir $SPECIES.$CHRNAME.STAR.Index --runThreadN $THREADS \
		--genomeFastaFiles $DATADIR/$SPECIES.$CHRNAME.fasta --genomeChrBinNbits 8 --genomeSAindexNbases 11
fi

for SRA in $srSRAS; do
	if [[ ! -s $SPECIES.RNA.$CHRNAME.$SRA.Aligned.out.bam ]]; then
		echo -e "###### `date`: Mapping $SRA reads onto the $SPECIES genome"
		echo 'STAR --genomeDir $SPECIES.$CHRNAME.STAR.Index --readFilesIn $DATADIR/${SRA}_1.extracted.fastq.gz $DATADIR/${SRA}_2.extracted.fastq.gz --outFilterType BySJout \'
		echo '   --outSAMattributes All --outSAMtype BAM Unsorted --readFilesCommand zcat \'
		echo '   --runThreadN 16 --alignEndsType Local --outStd Log --limitBAMsortRAM 7900000000 \'
		echo '   --alignSJoverhangMin 10 --outFileNamePrefix $SPECIES.RNA.$CHRNAME.$SRA.'
		time STAR --genomeDir $SPECIES.$CHRNAME.STAR.Index --readFilesIn $DATADIR/${SRA}_1.extracted.fastq.gz $DATADIR/${SRA}_2.extracted.fastq.gz --outFilterType BySJout \
			--outSAMattributes All --outSAMtype BAM Unsorted --readFilesCommand zcat \
			--runThreadN 16 --alignEndsType Local --outStd Log --limitBAMsortRAM 7900000000 \
			--alignSJoverhangMin 10 --outFileNamePrefix $SPECIES.RNA.$CHRNAME.$SRA.
fi
done

if [[ ! -s $SPECIES.RNA.$CHRNAME.Aligned.sortedByCoord.out.bam ]]; then
	echo 'ls -1 $SPECIES.RNA.$CHRNAME.*.Aligned.out.bam > BAM.list'
	ls -1 $SPECIES.RNA.$CHRNAME.*.Aligned.out.bam > BAM.list
	cat BAM.list

	echo 'samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.RNA.$CHRNAME.Aligned.merged.bam'
	time samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.RNA.$CHRNAME.Aligned.merged.bam
	echo 'samtools sort -@ 20 $SPECIES.RNA.$CHRNAME.Aligned.merged.bam > $SPECIES.RNA.$CHRNAME.Aligned.sortedByCoord.out.bam'
	time samtools sort -@ 20 $SPECIES.RNA.$CHRNAME.Aligned.merged.bam > $SPECIES.RNA.$CHRNAME.Aligned.sortedByCoord.out.bam
fi

for SRA in $longREADS; do
	if [[ ! -s $SPECIES.$SRA.lonrRNA.$CHRNAME.bam ]]; then
		echo -e "###### `date`: Mapping $SRA reads onto the $SPECIES genome"
		echo 'winnowmap -ax splice $CHR $DATADIR/$SRA.extracted.fastq.gz | samtools view -@ 30 -O BAM > $SPECIES.$SRA.lonrRNA.bam'
		time winnowmap -ax splice $CHR $DATADIR/$SRA.extracted.fastq.gz | samtools view -@ 30 -O BAM > $SPECIES.$SRA.lonrRNA.bam
		echo 'samtools view -@ 30 -O BAM $SPECIES.$SRA.lonrRNA.bam | samtools sort -@ 30 -n > $SPECIES.$SRA.lonrRNA.$CHRNAME.bam'
		samtools view -@ 30 -O BAM $SPECIES.$SRA.lonrRNA.bam | samtools sort -@ 30 -n > $SPECIES.$SRA.lonrRNA.$CHRNAME.bam
fi
done

if [[ ! -s $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam ]]; then
	echo 'ls -1 $SPECIES.*.lonrRNA.$CHRNAME.bam > BAM.list'
	ls -1 $SPECIES.*.lonrRNA.$CHRNAME.bam > BAM.list

	cat BAM.list

	echo 'samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam'
	time samtools merge -@ 20 -f -O BAM -b BAM.list $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam
	echo 'samtools sort -@ 20 $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam > $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam'
	time samtools sort -@ 20 $SPECIES.lonrRNA.$CHRNAME.Aligned.merged.bam > $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam
	echo 'samtools index -@ 20 $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam'
	samtools index -@ 20 $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam
fi


if [[ ! -s $SPECIES.MiniProt.hints.gff ]]; then
	echo 'miniprot --aln --gff --trans -t 20 --trans -G 50000 $CHR $SWISSPROTDB.fasta > $SPECIES.MiniProt.gff'
	miniprot --aln --gff --trans -t 20 --trans -G 50000 $CHR $SWISSPROTDB.fasta > $SPECIES.MiniProt.gff
	echo 'Miniprot2SplicedNucl.py -g $SPECIES.MiniProt.gff > $SPECIES.MiniProt.nt.fasta'
	Miniprot2SplicedNucl.py -g $SPECIES.MiniProt.gff > $SPECIES.MiniProt.nt.fasta
	echo 'minimap2 -t 40 -ax splice:hq $CHR $SPECIES.MiniProt.nt.fasta > $SPECIES.MiniProt.nt.sam'
	minimap2 -t 40 -ax splice:hq $CHR $SPECIES.MiniProt.nt.fasta > $SPECIES.MiniProt.nt.sam
	echo 'sam2psl.py -i $SPECIES.MiniProt.nt.sam -s -o $SPECIES.MiniProt.psl'
	sam2psl.py -i $SPECIES.MiniProt.nt.sam -s -o $SPECIES.MiniProt.psl
	echo 'cat $SPECIES.MiniProt.psl | sort -n -k 16,16 | sort -s -k 14,14 > $SPECIES.MiniProt.sorted.psl'
	cat $SPECIES.MiniProt.psl | sort -n -k 16,16 | sort -s -k 14,14 > $SPECIES.MiniProt.sorted.psl
	echo 'blat2hints.pl --nomult --in=$SPECIES.MiniProt.sorted.psl --out=$SPECIES.MiniProt.hints.gff'
	blat2hints.pl --nomult --in=$SPECIES.MiniProt.sorted.psl --out=$SPECIES.MiniProt.hints.gff
	wc -l $SPECIES.MiniProt.hints.gff
fi
