# *Ab Initio* transcriptome assembly using long-reads with Stringtie and IsoQuant
In recent years, in contrast to traditional short-read sequencing methods, long-read sequencing has emerged as a powerful tool in transcriptome analysis, sequencing offers unique advantages in identifying full-length transcripts and uncovering previously overlooked biological features.
You can see the two approaches a bit like solving a very complex jigsaw puzzle (the transcriptome) with a lot of very small pieces compared with larger pieces.

<img width="1686" alt="SmallVsLarge" src="https://github.com/francicco/GenomeAnnotationWorkshop2024/assets/9006870/c01b30c4-6cc5-45c3-a31e-a9d1764e5cac">

Although long-read sequencing significantly enhances transcriptome profiling, promising accuracy, and full-length RNAs, things are not so straight forward.

![image](https://github.com/francicco/GenomeAnnotationWorkshop2024/assets/9006870/9657aa7c-d119-4d5e-8b22-7e8407466f42)
from (https://github.com/ablab/IsoQuant)

Bioinformatics methods applied to transcriptome assembly reconstruction using long-read sequencing are quickly growing in number. Here we present two of the methods that at the moment perform better. These are [`StringTie`](https://www.nature.com/articles/nbt.3122) (which you've seen applied on short-read data) and [`IsoQuant`](https://www.nature.com/articles/s41587-022-01565-y), which can also jointly use short-read data to further refine long-read signal.

## StringTie & IsoQuant pipelines
In this practical, we will use a very similar approach we used for short-read data, which means to execute the main program and run the other utilities for exploring the results.

### Run `StringTie` as you did before using as input a BAM file:
```bash
stringtie $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam -L \
	-o $SPECIES.lonrRNA.$CHRNAME.stringtie.out.gtf -p $THREADS -m 100 $STRAND
```
Note: as an alternative you can include `-G $BRAKERGTF` to run the analysis in a guided fashion.

Now we convert the `gtf` into a `bed12` file...
```bash
mikado util convert -if gtf -of bed12 $SPECIES.lonrRNA.$CHRNAME.stringtie.out.gtf | sed 's/;coding=False//' | sed 's/ID=//' > $SPECIES.lonrRNA.$CHRNAME.stringtie.out.bed
```

...extract the relative nucleotide sequences...
```bash
bedtools getfasta -name -fo $SPECIES.lonrRNA.$CHRNAME.stringtie.out.fasta -fi $DATADIR/$SPECIES.$CHRNAME.fasta -bed $SPECIES.lonrRNA.$CHRNAME.stringtie.out.bed
```

...run `BUSCO`, 
```bash
busco -f -i $SPECIES.lonrRNA.$CHRNAME.stringtie.out.fasta --cpu $THREADS -m transcriptome -l $BUSCODIR/$BUSCODB --out run_$SPECIES.lonrRNA.$CHRNAME.stringtie.out.$BUSCODB
```

and `diamond blastx`,
```bash
 diamond blastx  --ultra-sensitive --max-target-seqs 1 --threads $THREADS --query $SPECIES.lonrRNA.$CHRNAME.stringtie.out.fasta --outfmt 6 --db ${SWISSPROTDB} \
		--evalue 1e-5 --out $SPECIES.lonrRNA.$CHRNAME.stringtie.out.outfmt6
```

to explore possible fragmentation and completness.
```bash
analyze_blastPlus_topHit_coverage.pl $SPECIES.lonrRNA.$CHRNAME.stringtie.out.outfmt6 $SPECIES.lonrRNA.$CHRNAME.stringtie.out.fasta ${SWISSPROTDB}.fasta
grep -v '^#' $SPECIES.lonrRNA.$CHRNAME.stringtie.out.outfmt6.w_pct_hit_length | sed 's/^/Scallop\t/' > $SPECIES.lonrRNA.$CHRNAME.stringtie.out.outfmt6.w_pct_hit_length.tsv
```


### I suggest to run `IsoQuant` with a reference annotation, it should perform better, and it should correct possible incongruences using signal from long-read data.

But first we need to remove few lines from the BRAKER annotation that will cause reading problems
```bash
grep -vw 'CDS\|5UTR\|3UTR\|start_codon\|stop_codon' $BRAKERGTF > $SPECIES.BRAKER.isoquant.gff3
```

Now run `IsoQuant`. It has some different flags such as `-d`, which needs to be used according to the type of long-reads you have.
```bash
isoquant.py -d nanopore --report_novel_unspliced true --stranded none --threads $THREADS --genedb $SPECIES.BRAKER.isoquant.gff3 \
  --bam $SPECIES.lonrRNA.$CHRNAME.sortedByCoord.out.bam --reference $CHR --output $SPECIES.IsoQuant \
  --model_construction_strategy all --check_canonical --prefix ${SPECIES}IsoSeq --data_type $LONGTYPE \
  --check_canonical --splice_correction_strategy all --model_construction_strategy all --count_exons
```

I provided a script I made to convert the `gtf` into a bed, it works a bit better to `Mikado`.
```bash
IsoQuantGTF2BED12v0.1.py -g ./$SPECIES.IsoQuant/${SPECIES}IsoSeq/${SPECIES}IsoSeq.extended_annotation.gtf > $SPECIES.RNA.$CHRNAME.isoquant.out.bed
```

The following is the same...
```bash
bedtools getfasta -name -fo $SPECIES.RNA.$CHRNAME.isoquant.out.fasta -fi $DATADIR/$SPECIES.$CHRNAME.fasta -bed $SPECIES.RNA.$CHRNAME.isoquant.out.bed
```

`BUSCO` for completeness...
```bash
busco -m transcriptome -i $SPECIES.RNA.$CHRNAME.isoquant.out.fasta -f -o run_$SPECIES.lonrRNA.$CHRNAME.isoquant.out.$BUSCODB -l $BUSCODIR/$BUSCODB --cpu $THREADS
```

then `Diamond`
```bash
diamond blastx  --ultra-sensitive --max-target-seqs 1 --threads $THREADS --query $SPECIES.RNA.$CHRNAME.isoquant.out.fasta --outfmt 6 --db ${SWISSPROTDB} \
		--evalue 1e-5 --out $SPECIES.RNA.$CHRNAME.isoquant.out.outfmt6
```

and `analyze_blastPlus_topHit_coverage.pl`
```bash
analyze_blastPlus_topHit_coverage.pl $SPECIES.RNA.$CHRNAME.isoquant.out.outfmt6 $SPECIES.RNA.$CHRNAME.isoquant.out.fasta ${SWISSPROTDB}.fasta
grep -v '^#' $SPECIES.RNA.$CHRNAME.isoquant.out.outfmt6.w_pct_hit_length | sed 's/^/IsoQuant\t/' > $SPECIES.RNA.$CHRNAME.isoquant.out.outfmt6.w_pct_hit_length.tsv
```

Merge the two or more tables, if you have more, and plot the data
```bash
head -n1 $DATADIR/$SPECIES.Chr1.proteins.*.outfmt6.w_pct_hit_length > $SPECIES.AbInitioLongReadsVsRef.w_pct_hit_length.tsv
cat $SPECIES.RNA.$CHRNAME.stringtie.out.outfmt6.w_pct_hit_length.tsv $SPECIES.RNA.$CHRNAME.isoquant.out.outfmt6.w_pct_hit_length.tsv >> $SPECIES.AbInitioLongReadsVsRef.w_pct_hit_length.tsv
Rscript ~/software/bioscript/Analyze_Diamond_topHit_coverage.R $SPECIES.AbInitioLongReadsVsRef.w_pct_hit_length.tsv $SPECIES.AbInitioLongReadsVsRef.w_pct_hit_length.png
```
